src/ontoweaver/transformer.py

```diff
@@ -70,11 +94,10 @@ def __call__(self, row, i):
             str: Each split item from the cell value.
         """
         for key in self.columns:
-            if self.valid(row[key]):
-                assert(type(row[key]) == str)
+            if self.output_validator(pd.DataFrame([row[key]], columns=["cell_value"])):
                 items = row[key].split(self.separator)
                 for item in items:
-                    if self.valid(item):
+                    if self.output_validator(pd.DataFrame([str(item)], columns=["cell_value"])):
```

Comment:
I think this is too complicated to ask a transformer developer to systematically think about using that.
Would it be possible to add a function in the interface that masks that?
Something like:
```suggestion
                    self.create(item)

    def create(self, item, node_type = None, edge_type = None):
                    res = str(item)
                    if self.output_validator(pd.DataFrame([res], columns=["cell_value"])):
                        yield res
                    else:
                        msg = f"Transformer {self.__name__} did not produced valid data {self.output_validator.message}"
                        logging.error(msg)
                        raise DataValidationError(msg)
```
This would allow for cleaner error management, and be extensible to the foreseen node/edge types management.

Reply:
This was originally the plan in the comment I made about having a separate function that would check and yield the result, but I was having a hard time finding a way for it to be general enough to accommodate all of the transformer outputs, since some of the transformers do additional processing steps, like for example `replace` or `cat_format`.